lib/broadway.ex

```diff
@@ -897,8 +912,8 @@ defmodule Broadway do
       [MyBroadway.Producer_0, MyBroadway.Producer_1, ..., MyBroadway.Producer_7]
 
   """
-  @spec producer_names(broadway :: atom()) :: [atom()]
-  def producer_names(broadway) when is_atom(broadway) do
+  @spec producer_names(broadway :: atom() | via_tuple()) :: [atom()]
+  def producer_names(broadway) do
```

Comment:
Had to lose the guard clause for `is_atom()` with these functions. Lmk if there's a better option?

Reply:
You can do

```suggestion
  def producer_names(broadway) when is_atom(broadway) or (is_tuple(broadway) and tuple_size(broadway) == 3) do
```

and it already catches most cases of wrong stuff going in, IMO :)

Reply:
We can add a private guard for this maybe 

```elixir
defguardp is_broadway_name(...) ...
```

Reply:
I like the guard route!

Reply:
TIL!